src/journal/journalctl.c

```diff
@@ -2552,7 +2561,7 @@ static int run(int argc, char *argv[]) {
                                         break;
                         }
 
-                        if (arg_until_set && !arg_reverse) {
+                        if (arg_until_set && !arg_reverse && arg_lines >= 0) {
```

Comment:
Is this change necessary?? It seems this change breaks `--until` _without_ `--lines`.

Reply:
Ouch, I got this inverted. It should be `&& arg_lines < 0` because otherwise the counter will get us covered.